experiment_manager.py

```diff
@@ -351,18 +365,26 @@ def sweep(self, config):
         outer_variables = Config(*variable_config.param_list[:-1])  # "outer" means all but the inner-most loop
         # N-dimensional loop with itertools.product: # (actually N-1 )
 
+        counter = 1
+        num_jobs = 1
+        for var in outer_variables.param_list:
+            num_jobs=num_jobs*len(var.value)
 
         for indices, vals in enumerated_product(*outer_variables.get_values()):
```

Comment:
I think `enumerated_product(*outer_variables.get_values())` is used in enough places to be a function. maybe a method of `Config` but i am not sure

Reply:
I agree but this requires more thought because outer_variables is also used inside the loop.
I leave it for now but I add a TODO and it will be dealt with at some point..